services/api/db/views.py

```diff
@@ -1,3 +1,105 @@
-from django.shortcuts import render
+from rest_framework.decorators import action
+from rest_framework.viewsets import ViewSet
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.permissions import IsAdminUser
+from db.models import Profile, OrganizationSettings
+from db.userSerializers import ProfileSerializer, OrganizationSettingsSerializer
+from db.renderer import UserRenderer
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.exceptions import NotFound, ValidationError
+from django.core.exceptions import ObjectDoesNotExist
 
-# Create your views here.
+
+
+class ProfileViewSet(ViewSet):
+    serializer_class = ProfileSerializer
+    renderer_classes = [UserRenderer]
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        try:
+            user_profile = Profile.objects.get(user=self.request.user)
+            return user_profile
+        except ObjectDoesNotExist:
```

Comment:
```suggestion
        except Profile.DoesNotExist:
```

https://docs.djangoproject.com/en/5.0/ref/models/querysets/#django.db.models.query.QuerySet.get

Reply:
resolved